src/grafter/matcha/alpha.clj

```diff
@@ -198,7 +198,7 @@
      (concat
       (mapv parse-pattern-row requireds)
       (when (seq optionals)
-        [`(l/conda
+        [`(l/conde
```

Comment:
Ok, so I'm assuming that this is the root source of the bug (with the rest of the changes) being to clean up the  implications this has on the results.

To be honest I'd thought we were doing a `l/conde`, and I can't think why this was implemented as a `l/conda`.  Though I looked and it seems we're wrapping the `l/conde` with an `l/conda`, so the `l/conde` I thought we were doing is at a different level.

I've checked in the history and it seems that originally I'd [said `OPTIONAL` was just a `conde`](https://github.com/Swirrl/matcha/issues/11#issuecomment-425566212) and that @andrewmcveigh had [originally implemented it as such here](https://github.com/Swirrl/matcha/pull/16/files#diff-cc4d6676244bf1219ff935b86132a16670d4d9e18fec723907532fcc855c4055R161) but that it was [later wrapped with a `conda`](https://github.com/Swirrl/matcha/commit/ebd019badb615ba3c22041522ae9499b52739252#diff-cc4d6676244bf1219ff935b86132a16670d4d9e18fec723907532fcc855c4055R185) as part of improvements to the parser/validator, and this is what you've changed to a `conde`

Now this comes up I do vaguely recall discussing this in person with @andrewmcveigh and us both feeling it wasn't quite right, but for some reason (probably the reason for this patch) `conde` didn't quite work, and that `conda` at the time seemed to give correct results, though I have very little recollection of the details, and obviously didn't give it enough attention at the time.

I'm also curious if the optionals worked properly prior to the parser/validator changes, and if this was a regression.